src/parametrization_tendencies.jl

```diff
@@ -1,13 +1,83 @@
-struct parametrization_tendencies{T<:AbstractFloat}
+struct parametrization_tendencies{NF<:AbstractFloat
 
 #Just a placeholder
 
+temp_convection ::Array{Complex{NF},3} #temperature tendency due to convection, grid point space
+humid_convection ::Array{Complex{NF},3} #humidity tendency due to convection, grid point space
+
 end
 
 """
 Compute physical parametrization tendencies
 """
-function parametrization_tendencies{T}(utend, vtend, ttend, qtend) where {T<:AbstractFloat}
+function parametrization_tendencies!(Prog::PrognosticVariables{NF}, # Prognostic variables
+                                     Diag::PrognosticVariables{NF},# Diagnostic variables
+                                     C::Constants{NF}
+                                     ) where {T<:AbstractFloat}
+
+
+    #Unpack relevant variables
+    @unpack temp_tend, humid_tend,u_tend,v_tend = Diag.Tendencies
+    @unpack pres_surf_grid,humid_grid,temp_grid,geopot_grid = Diag.gridvars
+
+    nlat,nlon,nlev = size(humid_grid)
+
+
+    # 1. Compute the thermodynamic variables
+
+    # 1.1 Declare some surface pressure quantities
+    pres_surf_grid_exp = exp(pres_surf_grid)
+    pres_surf_grid_exp_inv = 1/pres_surf_grid_exp
+
+    #1.2 Remove negative humidity grid values
+    #TK: don't really want to iterate in this way. Is there a nice inbuilt function to do the same thing?
```

Comment:
I'll add the following to `src/utils.jl`, so you can call `clip_negatives(humid_grid)` here. My Julia gut feeling tells me this is as fast as it gets

```julia
function clip_negatives(A::AbstractArray{T}) where T
    @inbounds for i in eachindex(A)
        A[i] = max(A[i],zero(T))
    end
end
```

Reply:
With #34 you can just write `cip_negatives!(humid_grid)` 😄

Reply:
Thats great!